src/System.Text.Json/docs/writable_json_dom_spec.md

```diff
@@ -0,0 +1,257 @@
+# Writable JSON Document Object Model (DOM) for `System.Text.Json`
+
+## Introduction
+
+`JsonNode` is a modifiable, dictionary-backed API to complement the readonly `JsonDocument`.
+
+It is the base class for the following concrete types representing all possible kinds of JSON nodes:
+* `JsonString` - representing JSON text value
+* `JsonBoolean` - representing JSON boolean value (`true` or `false`)
+* `JsonNumber` - representing JSON numeric value, can be created from and converted to all possible built-in numeric types
+* `JsonArray` - representing the array of JSON nodes
+* `JsonObject` - representing the set of properties - named JSON nodes
+
+It is a summer internship project being developed by @kasiabulat.
+
+## Goals
+
+The user should be able to:
+* Build up a structured in-memory representation of the JSON payload. 
+* Query the document object model.
+* Modify it. That includes, remove, add, and update. This means we want to build a modifiable JsonDocument that is not just readonly.
```

Comment:
It's a modifiable JsonDocument analogue, not a modifiable JsonDocument (since JsonDocument will stay readonly with the approach being pursued)